### PR TITLE
Design-2

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: queueUsingStack/main.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		this.stack1 = newHead
+		return
+	}
+	newHead.Next = this.stack1
+	this.stack1 = newHead
+}
+
+/*
+	time: o(n) -- because of flipStack1IntoStack2()
+	space: o(n) -- because of flipStack1IntoStack2()
```
Comment (amritgill27195) [inline queueUsingStack/main.go:71]: amortized o(1) for both time and space -- if stack2 already has items to process -- then the flip wont happen

File: queueUsingStack/main.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		this.stack2 = newHead
+		return out
+	}
+	return -1
+}
+
+/*
+	time: o(n) -- because of flipStack1IntoStack2()
+	space: o(n) -- because of flipStack1IntoStack2()
+*/
```
Comment (amritgill27195) [inline queueUsingStack/main.go:88]: amortized o(1) for both time and space -- if stack2 already has items to process -- then the flip wont happen

**Review by keshav-Gupta-IS — COMMENTED** [review]
Good Work, Amrit!! Keep it up.

**Comment by keshav-Gupta-IS** [comment]
Good Work, Amrit!! Keep it up.

**Pull request closed without merging**